extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/SegmentLoadStatusFetcher.java

```diff
@@ -149,46 +148,46 @@ public void waitForSegmentsToLoad()
     try {
       FutureUtils.getUnchecked(executorService.submit(() -> {
         long lastLogMillis = -TimeUnit.MINUTES.toMillis(1);
-        try {
-          while (!(hasAnySegmentBeenLoaded.get() && versionLoadStatusReference.get().isLoadingComplete())) {
-            // Check the timeout and exit if exceeded.
-            long runningMillis = new Interval(startTime, DateTimes.nowUtc()).toDurationMillis();
-            if (runningMillis > TIMEOUT_DURATION_MILLIS) {
-              log.warn(
-                  "Runtime[%d] exceeded timeout[%d] while waiting for segments to load. Exiting.",
-                  runningMillis,
-                  TIMEOUT_DURATION_MILLIS
-              );
-              updateStatus(State.TIMED_OUT, startTime);
-              return;
-            }
+        while (true) {
+          if (DateTimes.nowUtc().getMillis() - startTime.getMillis() > TIMEOUT_DURATION_MILLIS) {
+            log.warn("Timed out waiting for segments to load");
+            break;
+          }
 
-            if (runningMillis - lastLogMillis >= TimeUnit.MINUTES.toMillis(1)) {
-              lastLogMillis = runningMillis;
-              log.info(
-                  "Fetching segment load status for datasource[%s] from broker",
-                  datasource
-              );
+          try {
+            SqlQuery sqlQuery = new SqlQuery(
+                StringUtils.format(LOAD_QUERY, datasource, versionsConditionString),
+                ResultFormat.ARRAY,
+                false,
+                false,
+                false,
+                null,
+                null
+            );
+            
+            SqlTaskStatus taskStatus = FutureUtils.getUnchecked(brokerClient.submitSqlTask(sqlQuery), true);
+            if (taskStatus.getState() == TaskState.SUCCESS) {
```

Comment:
It is very unlikely that this `taskStatus` will ever be success or even failed. The task would have just been submitted to the Overlord.

Moreover, this use case doesn't require submitting an MSQ task to the Overlord since the LOAD_QUERY is just a sys query.